vmm/src/lib.rs

```diff
@@ -2011,77 +2002,31 @@ impl Vmm {
             })
     }
 
-    fn send_response(outcome: VmmRequestOutcome, sender: OutcomeSender) {
-        sender
-            .send(outcome)
-            .map_err(|_| ())
-            .expect("one-shot channel closed");
-    }
-
-    fn run_vmm_action(&mut self) -> Result<()> {
+    fn run_vmm_action(&mut self, vmm_action: VmmAction) -> VmmRequestOutcome {
```

Comment:
It looks like this function (together with the `VmmAction` enum and related logic) can be taken out of the vmm as well, since the driver can simply invoke methods such as `configure_boot_source`, `init_logger`, `flush_metrics`, `start_microvm`, etc. directly on a `Vmm` object.